### ISSUE TITLE
No support for Mqttv5 message expiry

### ISSUE THREAD

**Issue by iSchluff** [eclipse-paho/paho.mqtt.cpp#523]
Hi, as far as I am aware there is currently no way to pass the v5 message expiry option to the underlying C library?

Because the properties would need to be passed to the MQTTAsync_sendMessage function, but currently there is no signature allowing this.

E.g.
https://github.com/eclipse-paho/paho.mqtt.c/blob/6b1e202a701ffcdaa277b5644ed291287a70a7aa/src/samples/paho_c_pub.c#L352

But currently the cpp publish path looks like this
https://github.com/eclipse-paho/paho.mqtt.cpp/blob/3631bce7e0bd74cc11a46410669ccd2e23b6caed/src/async_client.cpp#L642-L652

**Comment by fpagliughi** [comment]
Yeah, this was definitely an oversight on my part when I added the v5 support, but you _can_ set message properties. You just need to create a `message` object with properties, then send it using the method:

    delivery_token_ptr publish(const_message_ptr msg) override;


To create the message with properties you can use a constructor:

    message(
        string_ref topic, const void* payload, size_t len, int qos, bool retained,
        const properties& props = properties()
    );

    message(
        string_ref topic, binary_ref payload, int qos, bool retained,
        const properties& props = properties()
    );

...or the equivalent `create()` methods to get the shared pointer directly.

Alternately, you can use the `message_ptr_builder` object to build a message with the properties you want.

This is clearly sub-optimal, and I will look at some additional message constructors and publish() overloads.

**Comment by fpagliughi** [comment]
Done. A few more message creation functions and `publish()` overloads take an optional set of properties. I'll add some code to an example showing how to set message expiry.

**Comment by iSchluff** [comment]
Thanks a lot for the changes
